src/main/java/com/develhope/spring/deals/services/OrderService.java

```diff
@@ -32,10 +33,21 @@ public OrderResponseDTO create(OrderCreatorDTO orderCreatorDTO) {
         return orderMapper.toResponseDTO(savedOrder);
     }
 
+    public void delete(Long orderId) {
+        if (!orderRepository.existsById(orderId)) {

```

Comment:
O catturi l'eccezione NoSuchElementException in u ntry catch e rilanci questa, oppure rimuovi questo controllo perchè già viene fatto dalla repository